### PR TITLE
Add PostSync Hook option to change Argo CD duration reported

### PR DESCRIPTION
**Pull request by andrea-mosk**
### What does this PR do? What is the motivation?

Two main things:
1. Improves the current Argo CD documentation by merging the three steps to create the setup into a single one.
2. Add a section to mention how to improve the reported Argo CD duration by taking resources start up into account.

### Merge instructions

Merge readiness:
- [x] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/continuous_delivery/deployments/argocd.md [hunk 1]
```diff
@@ -30,12 +30,11 @@ Argo CD notifications consists of two main components:
 
 ## Setup
 
-For more information on how to set up Argo CD notifications using webhooks, see the [official Argo CD guide][5].
+The setup below uses the [Webhook notification service][5] of Argo CD in order to send notifications to Datadog.
```
Comment (jhgilbert) [inline content/en/continuous_delivery/deployments/argocd.md:33]:
```suggestion
The setup below uses the [Webhook notification service][5] of Argo CD to send notifications to Datadog.
```

**Review by jhgilbert — APPROVED** [review]
Approved with a very minor suggestion, thank you!

File: content/en/continuous_delivery/deployments/argocd.md [hunk 2]
```diff
@@ -230,6 +230,10 @@ You can omit the `--config-repo` option if the CI is checked out to the configur
 
 If the command has been correctly run, deployments contain Git metadata from the application repository instead of the configuration repository. Also, the deployment executions view now contains a new **Pipeline** tab representing the related CI pipeline trace.
 
+## Visualize deployments in Datadog
+
+The [**Deployments**][6] and [**Executions**][7] pages populate with data after a deployment is executed. For more information, see [Search and Manage][9] and [CD Visibility Explorer][10].
```
Comment (alxch-) [inline content/en/continuous_delivery/deployments/argocd.md:235]:
```suggestion
The [**Deployments**][6] and [**Executions**][7] pages populate with data after a deployment has finished. For more information, see [Search and Manage][9] and [CD Visibility Explorer][10].
```

I think that executed is too broad, it does not say that data comes only when the deployment has finished.

File: content/en/continuous_delivery/deployments/argocd.md [hunk 3]
```diff
@@ -175,11 +153,33 @@ metadata:
...
+## Change duration to wait for resources health
+The duration reported in deployment events matches the sync duration in Argo CD. However, the sync duration generally represents the time spent by Argo CD to sync the Git repository state and the Kubernetes cluster state.
+This means that what happens after the sync (for example, the time spent by the Kubernetes resources to start up), is not included in the duration.
 
-## Visualize deployments in Datadog
+To change the duration reported to wait until the configured resources have started up and reached a healthy state, you can add a [PostSync Hook][19] to the resources that your Argo CD application monitors.
+The PostSync Hook will run after all the resources have reached a Healthy state, and the Argo CD sync will wait on the PostSync Hook result to update the application status as Healthy.
 
-The [**Deployments**][6] and [**Executions**][7] pages populate with data after a deployment is executed. For more information, see [Search and Manage][9] and [CD Visibility Explorer][10].
+Below is represented an example of a PostSync Hook Job that runs a simple `echo` command.
```
Comment (alxch-) [inline content/en/continuous_delivery/deployments/argocd.md:163]: People have to add a noop container in order to get the hook to be sent after the application is healthy? Is there any other alternative? I feel it's odd to just say that this is an example without explaining what else they could do (or not do)

**Review by gturbat — APPROVED** [review]

File: content/en/continuous_delivery/deployments/argocd.md [hunk 4]
```diff
@@ -113,75 +90,57 @@ metadata:
...
-- `tags.datadoghq.com/service` (required): specifies the Datadog service of this resource. For more information, see [Unified Service Tagging][18].
-- `team` (optional): specifies the Datadog team of this resource. If this label is omitted, the team is automatically retrieved from [Service Catalog][13] based on the service label.
+### Change duration to wait for resources health
+The duration reported in deployment events matches the sync duration in Argo CD. However, the sync duration generally represents the time spent by Argo CD to sync the Git repository state and the Kubernetes cluster state.
+This means that what happens after the sync (for example, the time spent by the Kubernetes resources to start up) is not included in the duration.
 
-Only the Kubernetes resources with the following kinds are eligible: `Deployment`, `ReplicaSet`, `StatefulSet`, `Service`, `DaemonSet`, `Pod`, `Job`, and `CronJob`.
+To change the duration reported to wait until the configured resources have started up and reached a healthy state, you can add a [PostSync Hook][19] to the resources that your Argo CD application monitors.
```
Comment (alxch-) [inline content/en/continuous_delivery/deployments/argocd.md:119]:
```suggestion
To change the duration reported to wait until the configured resources have started up and reached a healthy state, add a new no-op resource monitored by your Argo CD application, with a [PostSync Hook][19] annotation.
```

**Pull request merged**